docs/reference/pyproject-settings.rst

```diff
@@ -26,6 +26,17 @@ use them and examples.
    ``-Dpython.allow_limited_api=false`` option is passed to ``meson
    setup``.
 
+.. option:: tool.meson-python.meson
+
+   A string identifying a custom version of ``meson``. It may be an executable
+   name, or a path to an executable or script. If it's a path to a file that
+   exists and the string ends in ``.py``, then it will be invoked as a script
+   with the same Python interpreter that is used to run meson-python itself.
+
+   If this setting is used, it will be used for all invocation of ``meson``.
+   It can be overridden by a ``$MESON`` environment variable (see
+   :ref:`reference-environment-variables`).
+
```

Comment:
I would avoid to write "version of meson" as it makes it look like what is specified is the version and not the executable. I like the wording in the env var description, and I think we can keep this a bit simpler without loss of information:
```
A string specifying the ``meson`` executable or script to use. If is it is a path to
an existing file with a name ending in ``.py``, it will be invoked as a Python script
using the same Python interpreter that is used to run ``meson-python`` itself. It
can be overrrideen by the :envvar:`MESON` environment variable.
```
The ``:envvar:`MESON` `` thing should link to the right place in the documentation, if I got the syntax right.